tools/testing/gatekeeper/jobs.py

```diff
@@ -0,0 +1,250 @@
+
+
+class Checker:
+    """Object to keep watching required GH action workflows"""
+    def __init__(self):
+        self.latest_commit_sha = os.getenv("COMMIT_HASH")
+        self.pr_number = os.getenv("PR_NUMBER")
+        required_labels = os.getenv("REQUIRED_LABELS")
+        if required_labels:
+            self.required_labels = set(required_labels.split(";"))
+        else:
+            self.required_labels = []
+        required_jobs = os.getenv("REQUIRED_JOBS")
+        if required_jobs:
+            required_jobs = required_jobs.split(";")
+        else:
+            required_jobs = []
+        required_regexps = os.getenv("REQUIRED_REGEXPS")
+        self.required_regexps = []
+        # TODO: Add way to specify minimum amount of tests
+        # (eg. via \d+: prefix) and check it in status
+        if required_regexps:
+            for regexp in required_regexps.split(";"):
+                self.required_regexps.append(re.compile(regexp))
+        if not required_jobs and not self.required_regexps:
+            raise RuntimeError("No REQUIRED_JOBS or REQUIRED_REGEXPS defined")
+        # Set all required jobs as RUNNING to enforce waiting for them
+        self.results = {job: {"status": "EXPECTED", "run_id": -1}
+                        for job in required_jobs}
+
+    def record(self, workflow, job):
+        """
+        Records a job run
+        """
+        job_name = f"{workflow} / {job['name']}"
+        if job_name not in self.results:
+            for re_job in self.required_regexps:
+                # Required job via regexp
+                if re_job.match(job_name):
+                    break
+            else:
+                # Not a required job
+                return
+        # TODO: Check if multiple re-runs use the same "run_id". If so use
+        #       job['run_attempt'] in case of matching "run_id".
+        elif job['run_id'] <= self.results[job_name]['run_id']:
+            # Newer results already stored
+            print(f"older {job_name} - {job['status']} {job['conclusion']} "
+                  f"{job['id']}", file=sys.stderr)
+            return
+        print(f"{job_name} - {job['status']} {job['conclusion']} {job['id']}",
+              file=sys.stderr)
+        self.results[job_name] = job
+
+    @staticmethod
+    def _job_status(job):
+        """Map job status to our status"""
+        if job["status"] != "completed":
+            return RUNNING
+        if job["conclusion"] != "success":
+            return job['conclusion']
+        return PASS
+
+    def status(self):
+        """
+        :returns: 0 - all tests passing; 127 - no failures but some
+            tests in progress; 1 - any failure
+        """
+        running = False
+        if not self.results:
+            # No results reported so far
+            return FAIL
+        for job in self.results.values():
+            status = self._job_status(job)
+            if status == RUNNING:
+                running |= True
```

Comment:
Is there any particular reason for using `running |= True` instead of the more intuitive `running = True`?

Reply:
Not really, I'm just used to this from before as it (used to be) faster.